src/core/layout/qgslayoutitemlegend.h

```diff
@@ -44,11 +46,48 @@ class CORE_EXPORT QgsLegendModel : public QgsLayerTreeModel
 
   public:
     //! Construct the model based on the given layer tree
-    QgsLegendModel( QgsLayerTree *rootNode, QObject *parent SIP_TRANSFERTHIS = nullptr );
+    QgsLegendModel( QgsLayerTree *rootNode, QObject *parent SIP_TRANSFERTHIS = nullptr, QgsLayoutItemLegend *layout = nullptr );
+
+    //! Constructor used in QgsLayoutItemLegend, allows access to host class
```

Comment:
Docstrings don't normally mention where something is used. They help someone who uses it in the future to understand what the parameters mean.

Reply:
True, I'll simply state that this is an alternative constructor.